packages/core-data/src/queried-data/get-query-parts.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Internal dependencies
+ */
+import { withWeakMapCache } from '../utils';
+
+/**
+ * An object of properties describing a specific query.
+ *
+ * @typedef {WPQueriedDataQueryParts}
+ *
+ * @property {number} page      The query page (1-based index, default 1).
+ * @property {number} perPage   Items per page for query (default 10).
+ * @property {string} stableKey An encoded stable string of all non-pagination
+ *                              query parameters.
+ */
+
+/**
+ * Given a query object, returns an object of parts, including pagination
+ * details (`page` and `perPage`, or default values). All other properties are
+ * encoded into a stable (idempotent) `stableKey` value.
+ *
+ * @param {Object} query Optional query object.
+ *
+ * @return {WPQueriedDataQueryParts} Query parts.
+ */
+export function getQueryParts( query ) {
+	/**
+	 * @type {WPQueriedDataQueryParts}
+	 */
+	const parts = {
+		stableKey: '',
+		page: 1,
+		perPage: 10,
+	};
+
+	// Ensure stable key by sorting keys. Also more efficient for iterating.
+	const keys = Object.keys( query ).sort();
+
+	for ( let i = 0; i < keys.length; i++ ) {
+		const key = keys[ i ];
+		const value = query[ key ];
+
+		switch ( key ) {
+			case 'page':
+			case 'perPage':
+				parts[ key ] = value;
```

Comment:
Should `value` be mapped to string or number to ensure it has the same type when passing to the cache no matter what plugin developer provided as a value? Knowing the dynamic

Reply:
```js
	it( 'encodes stable string key with page normalized', () => {
		const first = getQueryParts( { '?': '&', b: 2, page: 1 } );
		const second = getQueryParts( { b: 2, '?': '&', page: '1' } );

		expect( first ).toEqual( second );
		expect( first ).toEqual( {
			page: 1,
			perPage: 10,
			stableKey: '%3F=%26&b=2',
		} );
	} );
```

This fails.

Reply:
Not certain about this, I'll defer to @aduth as he originally built this utility.

Reply:
Yes, it seems reasonable to `Number( value )` here.

Reply:
>Should `value` be mapped to string or number to ensure it has the same type when passing to the cache no matter what plugin developer provided as a value?

Normalized to number as of 37e0d7a.